Libraries/MiscDrivers/CODEC/max9867.c

```diff
@@ -0,0 +1,431 @@
+* in all copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+* OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+* IN NO EVENT SHALL MAXIM INTEGRATED BE LIABLE FOR ANY CLAIM, DAMAGES
+* OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+* ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+* OTHER DEALINGS IN THE SOFTWARE.
+*
+* Except as contained in this notice, the name of Maxim Integrated
+* Products, Inc. shall not be used except as stated in the Maxim Integrated
+* Products, Inc. Branding Policy.
+*
+* The mere transfer of this software does not imply any licenses
+* of trade secrets, proprietary technology, copyrights, patents,
+* trademarks, maskwork rights, or any other form of intellectual
+* property whatsoever. Maxim Integrated Products, Inc. retains all
+* ownership rights.
+*******************************************************************************/
+
+#include <stdint.h>
+#include "i2c.h"
+#include "max9867.h"
+
+// clang-format off
+#define MAX9867_00_STATUS            0x00
+#define MAX9867_01_JACKSENSE         0x01
+#define MAX9867_02_AUX_HIGH          0x02
+#define MAX9867_03_AUX_LOW           0x03
+#define MAX9867_04_INT_EN            0x04
+#define MAX9867_05_SYS_CLK           0x05
+#define MAX9867_06_CLK_HIGH          0x06
+#define MAX9867_07_CLK_LOW           0x07
+#define MAX9867_08_DAI_FORMAT        0x08
+#define MAX9867_09_DAI_CLOCK         0x09
+#define MAX9867_0A_DIG_FILTER        0x0A
+#define MAX9867_0B_SIDETONE          0x0B
+#define MAX9867_0C_LVL_DAC           0x0C
+#define MAX9867_0D_LVL_ADC           0x0D
+#define MAX9867_0E_LVL_LINE_IN_LEFT  0x0E
+#define MAX9867_0F_LVL_LINE_IN_RIGHT 0x0F
+#define MAX9867_10_VOL_LEFT          0x10
+#define MAX9867_11_VOL_RIGHT         0x11
+#define MAX9867_12_MIC_GAIN_LEFT     0x12
+#define MAX9867_13_MIC_GAIN_RIGHT    0x13
+#define MAX9867_14_ADC_INPUT         0x14
+#define MAX9867_15_MIC               0x15
+#define MAX9867_16_MODE              0x16
+#define MAX9867_17_PWR_SYS           0x17
+#define MAX9867_FF_REV_ID            0xFF
+// clang-format on
+
+#define MAX9867_ADDR 0x18
+#define MAX9867_REVID 0x42
+
+#define MHZ(N) (N*1000000)
+
+static mxc_i2c_req_t i2c_req;
+
+static int reg_write(uint8_t reg, uint8_t val)
+{
+  uint8_t buf[2] = { reg, val };
+
+  i2c_req.tx_buf = buf;
+  i2c_req.tx_len = sizeof(buf);
+  i2c_req.rx_len = 0;
+
+  return MXC_I2C_MasterTransaction(&i2c_req);
+}
+
+static int reg_read(uint8_t reg, uint8_t *dest)
+{
+  uint8_t buf[1] = { reg };
+
+  i2c_req.tx_buf = buf;
+  i2c_req.tx_len = sizeof(buf);
+  i2c_req.rx_buf = dest;
+  i2c_req.rx_len = 1;
+
+  return MXC_I2C_MasterTransaction(&i2c_req);
+}
+
+static int reg_update(uint8_t reg, uint8_t mask, uint8_t val)
+{
+  int err;
+  uint8_t tmp;
+
+  if ((err = reg_read(reg, &tmp)))
```

Comment:
Compare result to E_NO_ERROR

Reply:
Addressed in https://github.com/Analog-Devices-MSDK/msdk/pull/240/commits/1c5ae59594fba5650b3ca48105437a28f0cc9f0f